e2e/scenario_test.go

```diff
@@ -649,6 +649,31 @@ func Test_ubuntu2204(t *testing.T) {
 	})
 }
 
+func Test_ubuntu2204Scriptless(t *testing.T) {
+	RunScenario(t, &Scenario{
+		Description: "Tests that a node using the Ubuntu 2204 Scriptless VHD can be properly bootstrapped",
+		Config: Config{
+			Cluster: ClusterKubenet,
+			VHD:     config.VHDUbuntu2204Gen2Containerd,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-ubuntu-containerd-22.04-gen2"
+				nbc.AgentPoolProfile.Distro = "aks-ubuntu-containerd-22.04-gen2"
+				// Check that we don't leak these secrets if they're
+				// set (which they mostly aren't in these scenarios).
+				nbc.ContainerService.Properties.CertificateProfile.ClientPrivateKey = "client cert private key"
+				nbc.ContainerService.Properties.ServicePrincipalProfile.Secret = "SP secret"
+			},
+			LiveVMValidators: []*LiveVMValidator{
+				mobyComponentVersionValidator("containerd", getExpectedPackageVersions("containerd", "ubuntu", "r2204")[0], "apt"),
+				mobyComponentVersionValidator("runc", getExpectedPackageVersions("runc", "ubuntu", "r2204")[0], "apt"),
+			},
+		},
+		Tags: Tags{
+			Scriptless: true,
```

Comment:
Does that mean by setting this tag `Scriptless`, client like aks-rp can determine sending scriptless contract or the old CSE scripts at runtime? Or this tag is only for e2e test?

Reply:
No, this is just for e2e testing